### PR TITLE
Optimise the TR181 GET/SET handler in webconfig.

### PR DESCRIPTION
**Pull request by Sreedevi29**

### CODE REVIEW

File: src/webcfg_rbus.c [hunk 1]
```diff
@@ -245,12 +361,40 @@ rbusError_t webcfgDataSetHandler(rbusHandle_t handle, rbusProperty_t prop, rbusS
...
+
+rbusError_t webcfgFrSetHandler(rbusHandle_t handle, rbusProperty_t prop, rbusSetHandlerOptions_t* opts) {
+
+    WebcfgDebug("Inside webcfgFrSetHandler\n");
+    (void) handle;
+    (void) opts;
+    char const* paramName = rbusProperty_GetName(prop);
+    if(strncmp(paramName, WEBCFG_FORCESYNC_PARAM, maxParamLen) != 0)
+    {
+        WebcfgError("Unexpected parameter = %s\n", paramName); //free paramName req.?
```
Comment (sadhyama) [inline src/webcfg_rbus.c:383]: Please check if this need to be freed

File: src/webcfg_rbus.c [hunk 2]
```diff
@@ -478,14 +697,33 @@ rbusError_t webcfgDataGetHandler(rbusHandle_t handle, rbusProperty_t property, r
...
-	else if(strncmp(propertyName, WEBCFG_SUPPLEMENTARY_TELEMETRY_PARAM, maxParamLen)==0){
+rbusError_t webcfgTelemetryGetHandler(rbusHandle_t handle, rbusProperty_t property, rbusGetHandlerOptions_t* opts) {
+    
+    WebcfgDebug("In webcfgTelemetryGetHandler\n");
+    (void) handle;
+    (void) opts;
+    char const* propertyName;
+    rbusError_t retPsmGet = RBUS_ERROR_BUS_ERROR;
+
+    propertyName = strdup(rbusProperty_GetName(property));
```
Comment (sadhyama) [inline src/webcfg_rbus.c:713]: propertyName need to be freed. I think we need to add free for propertyName , paramName in multiple get/set handlers e.g. line 316, 383 etc.

**Review by sadhyama — CHANGES_REQUESTED** [review]

File: src/webcfg_rbus.c (hunk 1 above)
Comment (sadhyama) [reply src/webcfg_rbus.c:383]: If not required to free, remove this comment too

File: src/webcfg_rbus.c [hunk 3]
```diff
@@ -198,32 +240,106 @@ rbusError_t webcfgDataSetHandler(rbusHandle_t handle, rbusProperty_t prop, rbusS
...
+
+rbusError_t webcfgTelemetrySetHandler(rbusHandle_t handle, rbusProperty_t prop, rbusSetHandlerOptions_t* opts) {
+
+    WebcfgDebug("Inside webcfgTelemetrySetHandler\n");
+    (void) handle;
+    (void) opts;
+    char const* paramName = rbusProperty_GetName(prop);
+    if(strncmp(paramName, WEBCFG_SUPPLEMENTARY_TELEMETRY_PARAM, maxParamLen) != 0) 
+    {
+        WebcfgError("Unexpected parameter = %s\n", paramName); //free paramName req.?
```
Comment (sadhyama) [inline src/webcfg_rbus.c:316]: Remove this comment

File: src/webcfg_rbus.c [hunk 4]
```diff
@@ -245,12 +361,40 @@ rbusError_t webcfgDataSetHandler(rbusHandle_t handle, rbusProperty_t prop, rbusS
...
-	if (!RFC_ENABLE)
+    }   
+     WebcfgDebug("webcfgTelemetrySetHandler End\n");
+     return RBUS_ERROR_SUCCESS;
+}
+      
+
+rbusError_t webcfgFrSetHandler(rbusHandle_t handle, rbusProperty_t prop, rbusSetHandlerOptions_t* opts) {
+
+    WebcfgDebug("Inside webcfgFrSetHandler\n");
```
Comment (sadhyama) [inline src/webcfg_rbus.c:377]: Try to reduce this Inside , End debug logs in each function

**Review by sadhyama — CHANGES_REQUESTED** [review]

File: src/webcfg_rbus.c [hunk 5]
```diff
@@ -299,41 +443,47 @@ rbusError_t webcfgDataSetHandler(rbusHandle_t handle, rbusProperty_t prop, rbusS
...
 
     }
-    WebcfgDebug("webcfgDataSetHandler End\n");
+    WebcfgDebug("webcfgFrSetHandler End\n");
     return RBUS_ERROR_SUCCESS;
 }
 
 /**
- * Common data get handler for all parameters owned by Webconfig
+ * Data get handler for Webpa parameters
```
Comment (sadhyama) [inline src/webcfg_rbus.c:451]: Change it to Webconfig or remove this .

**Review by sadhyama — CHANGES_REQUESTED** [review]

File: src/webcfg_rbus.c [hunk 6]
```diff
@@ -478,29 +706,53 @@ rbusError_t webcfgDataGetHandler(rbusHandle_t handle, rbusProperty_t property, r
...
 		rbusValue_Release(value);
 		return 0;
 	}
-
         if(SupplementaryURLVal){
             rbusValue_SetString(value, SupplementaryURLVal);
 	}
         else{
 		retPsmGet = rbus_GetValueFromDB( WEBCFG_SUPPLEMENTARY_TELEMETRY_PARAM, &SupplementaryURLVal );
+                retPsmGet = RBUS_ERROR_BUS_ERROR;
```
Comment (sadhyama) [inline src/webcfg_rbus.c:748]: Remove this error code assigning for psm DB get.

**Review by sadhyama — APPROVED** [review]

File: src/webcfg_rbus.c [hunk 7]
```diff
@@ -75,6 +75,22 @@ bool isRbusEnabled()
...
+bool isRfcEnabled() 
+{
+        bool isRfc = false;
+	if(get_rbus_RfcEnable()) 
+	{
+		isRfc = true;
+	}
+	else
+	{
+		isRfc = false;
```
Comment (shilpa24balaji) [inline src/webcfg_rbus.c:87]: is this needed when default is false?

**Pull request merged**